internal/kubeclient/indexer.go

```diff
@@ -498,100 +499,47 @@ func indexStagesByWarehouse(obj client.Object) []string {
-}
-
-// IndexServiceAccountsByOIDCGroups sets up indexing of ServiceAccounts by
-// their OIDC group annotations.
-//
-// It configures the manager's field indexer to allow querying ServiceAccounts
-// using the ServiceAccountsByOIDCGroupIndexField selector.
-func IndexServiceAccountsByOIDCGroups(ctx context.Context, mgr ctrl.Manager) error {
-	return mgr.GetFieldIndexer().IndexField(
-		ctx,
-		&corev1.ServiceAccount{},
-		ServiceAccountsByOIDCGroupIndexField,
-		indexServiceAccountsByOIDCGroups,
-	)
-}
-
-// indexServiceAccountsByOIDCGroups is a client.IndexerFunc that indexes
-// ServiceAccounts by their OIDC group annotations.
-func indexServiceAccountsByOIDCGroups(obj client.Object) []string {
-	sa := obj.(*corev1.ServiceAccount) // nolint: forcetypeassert
-	rawGroups := strings.TrimSpace(sa.GetAnnotations()[rbacapi.AnnotationKeyOIDCGroups])
-	if rawGroups == "" {
-		return nil
-	}
-	groups := strings.Split(rawGroups, ",")
-	refinedGroups := make([]string, 0, len(groups))
-	for _, g := range groups {
-		if group := strings.TrimSpace(g); group != "" {
-			refinedGroups = append(refinedGroups, group)
+	rawClaimValues := []string{}
+	for annotationKey, annotationValue := range sa.GetAnnotations() {
+		if strings.HasPrefix(annotationKey, rbacapi.AnnotationKeyOIDCClaimNamePrefix) {
+			rawClaimValue := strings.TrimSpace(annotationValue)
+			if rawClaimValue == "" {
+				continue
+			}
+			rawClaimValues = append(rawClaimValues, rawClaimValue)
 		}
 	}
-	return refinedGroups
-}
-
-// IndexServiceAccountsByOIDCSubjects sets up indexing of ServiceAccounts by
-// their OIDC subject annotations.
-//
-// It configures the manager's field indexer to allow querying ServiceAccounts
-// using the ServiceAccountsByOIDCSubjectIndexField selector.
-func IndexServiceAccountsByOIDCSubjects(ctx context.Context, mgr ctrl.Manager) error {
-	return mgr.GetFieldIndexer().IndexField(
-		ctx,
-		&corev1.ServiceAccount{},
-		ServiceAccountsByOIDCSubjectIndexField,
-		indexServiceAccountsByOIDCSubjects,
-	)
-}
-
-// indexServiceAccountsByOIDCSubjects is a client.IndexerFunc that indexes
-// ServiceAccounts by their OIDC subject annotations.
-func indexServiceAccountsByOIDCSubjects(obj client.Object) []string {
-	sa := obj.(*corev1.ServiceAccount) // nolint: forcetypeassert
-	rawSubjects := strings.TrimSpace(sa.GetAnnotations()[rbacapi.AnnotationKeyOIDCSubjects])
-	if rawSubjects == "" {
+	if len(rawClaimValues) == 0 {
 		return nil
 	}
-	subjects := strings.Split(rawSubjects, ",")
-	refinedSubjects := make([]string, 0, len(subjects))
-	for _, s := range subjects {
-		if subject := strings.TrimSpace(s); subject != "" {
-			refinedSubjects = append(refinedSubjects, subject)
+	refinedClaimValues := []string{}
+	for _, rawClaimValue := range rawClaimValues {
+		claimValues := strings.Split(rawClaimValue, ",")
+		for _, e := range claimValues {
+			if email := strings.TrimSpace(e); email != "" {
+				refinedClaimValues = append(refinedClaimValues, email)
+			}
 		}
 	}
```

Comment:
This is the other side of [this](https://github.com/akuity/kargo/pull/2315/files#r1692044477)

Reply:
Fixed the indexer as above